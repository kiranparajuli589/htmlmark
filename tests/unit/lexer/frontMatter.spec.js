import {MarkdownParser} from "../../../lib/index.js";

const mdWithFrontMatter = `---
title: Hello World
---

# Hello World
`;

const mdWithoutFrontMatter = `# Hello World`

const mdWithArrayNumberAndObjectInFrontMatter = `---
title: Hello World
array: [1, 2, 3]
object: {a: 1, b: 2}
---

# Hello World
`
const mdp = new MarkdownParser({frontMatter: true})

describe('FrontMatter', function () {
	it('should parse front matter', function () {
		const parsed = mdp.parse(mdWithFrontMatter)
		expect(parsed).toMatchSnapshot()
	})
	it('should tokenize front matter', function () {
		const parsed = mdp.tokenize(mdWithFrontMatter)
		expect(parsed).toMatchSnapshot()
	})
	it('should tokenize front matter with array, number and object', function () {
		const parsed = mdp.tokenize(mdWithArrayNumberAndObjectInFrontMatter)
		expect(parsed).toMatchSnapshot()
	})
});
