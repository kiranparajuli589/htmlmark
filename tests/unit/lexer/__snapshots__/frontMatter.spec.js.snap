// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontMatter should parse front matter 1`] = `
"<h1>Hello World</h1>
"
`;

exports[`FrontMatter should tokenize front matter 1`] = `
Array [
  Object {
    "type": "front-matter",
    "value": Array [
      Object {
        "key": "title",
        "keyValue": "Hello World",
      },
    ],
  },
  Object {
    "indent": 0,
    "level": 1,
    "raw": "# Hello World",
    "setext": false,
    "tokens": Array [
      Object {
        "raw": "Hello World",
        "type": "text",
        "value": "Hello World",
      },
    ],
    "type": "heading",
    "value": "Hello World",
  },
  Object {
    "type": "new-line",
  },
]
`;

exports[`FrontMatter should tokenize front matter with array, number and object 1`] = `
Array [
  Object {
    "type": "front-matter",
    "value": Array [
      Object {
        "key": "title",
        "keyValue": "Hello World",
      },
      Object {
        "key": "array",
        "keyValue": Array [
          1,
          2,
          3,
        ],
      },
      Object {
        "key": "object",
        "keyValue": "{a: 1, b: 2}",
      },
    ],
  },
  Object {
    "indent": 0,
    "level": 1,
    "raw": "# Hello World",
    "setext": false,
    "tokens": Array [
      Object {
        "raw": "Hello World",
        "type": "text",
        "value": "Hello World",
      },
    ],
    "type": "heading",
    "value": "Hello World",
  },
  Object {
    "type": "new-line",
  },
]
`;
